lib/ansible/modules/source_control/bitbucket/bitbucket_pipeline_key_pair.py

```diff
@@ -0,0 +1,213 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2019, Evgeniy Krysanov <evgeniy.krysanov@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community',
+}
+
+DOCUMENTATION = r'''
+---
+module: bitbucket_pipeline_key_pair
+short_description: Manages Bitbucket pipeline SSH key pair
+description:
+  - Manages Bitbucket pipeline SSH key pair.
+version_added: "2.8"
+author:
+  - Evgeniy Krysanov (@catcombo)
+options:
+  client_id:
+    description:
+      - OAuth consumer key.
+      - If not set the environment variable C(BITBUCKET_CLIENT_ID) will be used.
+    type: str
+  client_secret:
+    description:
+      - OAuth consumer secret.
+      - If not set the environment variable C(BITBUCKET_CLIENT_SECRET) will be used.
+    type: str
+  repository:
+    description:
+      - The repository name.
+    type: str
+    required: true
+  username:
```

Comment:
opt for `user` over `username` unless there's a good reason to differ

Reply:
There are two reasons why we opted for `username` instead of `user` for these modules:

1. Bitbucket (like Github) uses `username` in the API & GUI to denote an account
2. An account (username) might be either a _team_ or a _user_, so `user` has a different meaning

We thought that following this pattern would cause least confusion; what we want to achieve is that one specifies the two parts identifying the repository: `{username}/{repository}`, where `username` might be a team name.

We can change it to `user`, but have a preference for `username`; let us know if you don't think that our arguments are strong enough and we should change it anyways.

Reply:
In my opinion, there are two reasons to leave as is:
1. Other Bitbucket modules (including https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/source_control/bitbucket/bitbucket_pipeline_variable.py already accepted) use the same variable name
2. Bitbucket API uses `username` as named part of API url for [key pair endpoint](https://developer.atlassian.com/bitbucket/api/2/reference/resource/repositories/%7Busername%7D/%7Brepo_slug%7D/pipelines_config/ssh/key_pair)

Moreover, I think `username` is better than `user` because it more accurate description of what we want get. Bitbucket `username` is obviously the nickname, whereas the `user` has a fuzzy meaning: it could be an email, nickname or maybe some user ID.

Reply:
@zyv We think the same way ;)

Reply:
The reason I brought it up: https://github.com/ansible/ansible/issues/51757

Though in this case it may not be a connection variable.

Reply:
Right, in this case the connection vars are actually `client_id` and `client_secret`, for those we picked standard names, like in `typetalk` or `azure` modules. Using those connection variables, you can modify repositories belonging to any `username` you have access to, be it a team or a user.